src/main/java/de/yuna/berlin/nativeapp/services/http/model/HttpRequest.java

```diff
@@ -0,0 +1,250 @@
+        return isContentType(ContentType.IMAGE_JPEG.value());
+    }
+
+    public boolean isContentTypePng() {
+        return isContentType(ContentType.IMAGE_PNG.value());
+    }
+
+    public boolean isContentTypeGif() {
+        return isContentType(ContentType.IMAGE_GIF.value());
+    }
+
+    public boolean isContentTypeMpeg() {
+        return isContentType(ContentType.AUDIO_MPEG.value());
+    }
+
+    public boolean isContentTypeMp4() {
+        return isContentType(ContentType.VIDEO_MP4.value());
+    }
+
+
+    public String bodyAsString() {
+        try {
+            if (body == null)
+                body = new String(exchange.getRequestBody().readAllBytes(), StandardCharsets.UTF_8);
+            return body;
+        } catch (Exception ignored) {
+            return null;
+        }
+    }
+
+    public TypeContainer bodyAsJson() {
+        return JsonDecoder.jsonTypeOf(bodyAsString());
+    }
+
+    public TypeMap queryParameters() {
+        if (queryParams == null) {
+            try {
+                String query = exchange.getRequestURI().getQuery();
+                queryParams = new TypeMap();
+                if (query != null) {
+                    String[] queryParamsArray = query.split("&");
+                    for (String param : queryParamsArray) {
+                        String[] keyValue = param.split("=");
+
+                        if (keyValue.length == 2) {
+                            String key = java.net.URLDecoder.decode(keyValue[0], StandardCharsets.UTF_8);
+                            String value = java.net.URLDecoder.decode(keyValue[1], StandardCharsets.UTF_8);
+                            queryParams.put(key, value);
+                        } else if (keyValue.length == 1) {
+                            String key = java.net.URLDecoder.decode(keyValue[0], StandardCharsets.UTF_8);
+                            queryParams.put(key, "");
+                        }
+                    }
+                }
+            }catch (Exception ignored) {
+            }
+        }
+        return queryParams;
+    }
+
+    public boolean containsQueryParam(String key) {
+        return queryParams.containsKey(key);
+    }
+
+    public String getQueryParam(String key) {
+        return queryParams.get(key).toString();
+    }
+
+
+    public boolean exactMatch(String path) {
+        return this.path.startsWith(path);
+    }
+
+    public boolean match(String pathToMatch) {
+
+        String[] partsToMatch = pathToMatch.split("/");
+        String[] parts = path.split("/");
+
+        if (exactMatch(pathToMatch) && pathToMatch.contains("{"))
+            return true;
+
+        if (partsToMatch.length != parts.length)
+            return false;
+
+        if (pathParams == null) {
+            pathParams = new TypeMap();
+            for (int i = 0; i < partsToMatch.length; i++) {
+                if (!partsToMatch[i].equals(parts[i])) {
+                    if (partsToMatch[i].startsWith("{")) {
```

Comment:
Wow, that's brilliant!. I wonder if we could have a helper method to get these path prams, as we have the key names in pathToMatch and have the values in `parts`.

Additionally, I am thinking if it is possible to do more quick matches like counting the slashes before making a whole path check, which could be also a really efficient value to cache for the `path`. Like path length.

```
private int countSlashes(String s) {
    return (int) s.chars().filter(c -> c == '/').count();
}
```

Or just save the length or the split itself `String[] parts = path.split("/");`

Reply:
Yes, I have cached them in pathParams variable.
Plus this method is also checking the length of both parts if not match will return false.

Reply:
pathParams needs to be cached for each url (aka pathToMatch) ;)